test/lib/random_utils.hh

```diff
@@ -186,4 +186,13 @@ preemption_check random_preempt() {
     };
 }
 
+__attribute__((no_sanitize("undefined")))
+inline int64_t d2t(double d) {
```

Comment:
nit: random_utils is not the best place to put this helper as it's not related to random number generation.
(Maybe it's a bit misleading, but this header is not a collection for random utils)